src/commands/register.ts

```diff
@@ -147,7 +154,15 @@ export default class ComponentRegister extends Command {
       }
       let build_args: string[] = [];
       if (service_config.build?.args) {
-        build_args = Object.entries(service_config.build?.args).map(([key, value]) => `${key}=${value}`);
+        const build_args_map: Dictionary<string> = service_config.build?.args || {};
+        for (const arg of flags.arg || []) {
+          const [key, value] = arg.split('=');
+          if (!value) {
+            throw new Error('--arg must be in the format key=value');
```

Comment:
It could help if this was specific about which key/value pair failed

Reply:
sure